Gems/PhysX/Code/Source/JointComponent.h

```diff
@@ -108,5 +116,9 @@ namespace PhysX
         JointMotorProperties m_motor;
         AzPhysics::JointHandle m_jointHandle = AzPhysics::InvalidJointHandle;
         AzPhysics::SceneHandle m_jointSceneOwner = AzPhysics::InvalidSceneHandle;
+
+        // Variable to keep track of the rigid bodies' entities involved in this joint
+        // and if they are enabled or disabled.
+        AZStd::unordered_map<AZ::EntityId, bool> m_rigidBodyEntityMap;
```

Comment:
unordered_map feels a little bit overkill to me for something that will contain either one or two entities.

Reply:
I went for simplicity here since the map naturally contains the `key, element` pair needed for this case, there is no need to create additional struct or use a tuple (which could be more complicated to use), and since the code it's in a base class it won't be fixed to a number of elements, so it can scale without needing further changes (in case in the future whatever new child joint class wants/needs to depend on more entities to enable/disable the joint for whatever reason).

I could change it to a `AZStd::map` instead to be a tree and safe some memory since the look up will always be fast due to the low number of elements, would you prefer that?